staging/src/k8s.io/client-go/examples/controller-ref-manager/main.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"os"
+
+	apps "k8s.io/api/apps/v1beta2"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	basecontroller "k8s.io/client-go/tools/controller"
+	// Uncomment the following line to load the gcp plugin (only required to authenticate against GKE clusters).
+	// _ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+
+	"k8s.io/client-go/examples/controller-ref-manager/controller"
+)
+
+var (
+	productionLabel         = map[string]string{"type": "production", "app": "demo"}
+	productionLabelSelector = labels.Set{"type": "production"}.AsSelector()
+)
+
+// RefManager normally used in controllers sync loop to adopt/release resources.
+// See https://github.com/kubernetes/community/blob/master/contributors/design-proposals/api-machinery/controller-ref.md
+// for more details.
+func main() {
+	kubeconfig := flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
```

Comment:
please add the master option as well because `BuildConfigFromFlags` outputs confusing warnings otherwise.